controller/src/beerocks/master/db/db.cpp

```diff
@@ -1895,6 +1895,10 @@ bool db::set_supported_channel_radar_affected(const std::string &mac, std::vecto
     }
     auto channels_count = channels.size();
     LOG(DEBUG) << " channels_count = " << int(channels_count);
+    if (channels_count < 1) {
+        LOG(ERROR) << "the given channel list must contain at least one value";
+        return false;
+    }
     auto it =
         find_if(std::begin(n->hostap->supported_channels), std::end(n->hostap->supported_channels),
                 [&](beerocks::message::sWifiChannel supported_channel) {
```

Comment:
It crashes because of the "find" on line 1905 that iterate over `n->hostap->supported_channels` and compare it to `*channels.begin()` which is the given channel list on the function.
I don't understand why we do this. Why do we compare it only to the given list first element?
I would expect to do loop within a loop to and compare all the channels on list A with all the channels on list B, that way the loop would not have been performed on zero size scenario.
Yes, your solution will fix the crash, but it seems to be a bandaid to bad and undocumented code.

Reply:
Moved this to a new issue #1414 because:
1) That code is already merged, and shouldn't be a part of this PR
2) This PR needs to stay focused on fixing the crush issue